[ANGLE] Clamp up to base level in Metal backend

roberto-apple opened the pull request:
#### 014f513357e8701983f97c471ebfdb916f27db23
<pre>
[ANGLE] Clamp up to base level in Metal backend
<a href="https://bugs.webkit.org/show_bug.cgi?id=301328">https://bugs.webkit.org/show_bug.cgi?id=301328</a>
<a href="https://rdar.apple.com/152647032">rdar://152647032</a>

Reviewed by Kimmo Kinnunen.

The Metal backend will crash (i.e., fail a Metal assertion) in a setup
where mipmap base level is greater than max level and can result in
attempting to create a texture view with a range of 0.
In this case, clamp up to the base level.

* LayoutTests/TestExpectations:
* Source/ThirdParty/ANGLE/src/libANGLE/renderer/metal/TextureMtl.mm:
(rx::TextureMtl::createViewFromBaseToMaxLevel):

Canonical link: <a href="https://commits.webkit.org/302240@main">https://commits.webkit.org/302240@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bddb7b0723f8c76ba20d995d97a07aaa3dda0db6

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/128426 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/696 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/39257 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/135820 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/79877 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/48ad8b2f-7d00-4d8a-b78f-e905e910bb62) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/639 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/570 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/97757 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/65658 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/b06b46d2-180c-4be2-b4f0-fbf99431d9cd) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/131374 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/453 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/115058 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/78367 "Found 2 new API test failures: WPE/TestWebsiteData:/webkit/WebKitWebsiteData/itp, WPE/TestWebKitWebContext:/webkit/WebKitWebContext/no-web-process-leak (failure)") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/cf395526-85fa-46d4-b75b-dcf6b3d246ac) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/417 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/33165 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/79103 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/108821 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/33649 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/138270 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/539 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/506 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/106298 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/578 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/111399 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/106110 "Found 1 new API test failure: WebKitGTK/TestWebsiteData:/webkit/WebKitWebsiteData/itp (failure)") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27039 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/444 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/29942 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/52861 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/590 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/485 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/544 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/548 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple reviewed (commented):
Do you need to update the onBaseMaxLevelsChanged condition?
```
  if (mState.getEffectiveBaseLevel() == mNativeTextureStorage->getBaseGLLevel() &&


        mState.getMipmapMaxLevel() == mNativeTextureStorage->getMaxSupportedGLLevel())
```

roberto-apple committed:
[ANGLE] Clamp up to base level in Metal backend
https://bugs.webkit.org/show_bug.cgi?id=301328
rdar://152647032

Reviewed by Kimmo Kinnunen.

The Metal backend will crash (i.e., fail a Metal assertion) in a setup
where mipmap base level is greater than max level and can result in
attempting to create a texture view with a range of 0.
In this case, clamp up to the base level.

* LayoutTests/TestExpectations:
* Source/ThirdParty/ANGLE/src/libANGLE/renderer/metal/TextureMtl.mm:
(rx::TextureMtl::createViewFromBaseToMaxLevel):

Canonical link: https://commits.webkit.org/302240@main
